In [2]:
import pandas as pd

In [7]:
prefetch_output = pd.read_csv('../test/test_data/prefetch_trial.csv')
common_name = pd.read_csv('../db/common_name.csv')

In [41]:
common_name.shape

(240, 6)

In [40]:
prefetch_output.shape

(5015, 24)

In [32]:
def std_name(name):
    word_list = name.split()
    #accession,region = word_list[0].split(':')

    sci_name = word_list[1]+' '+word_list[2]
    other = ' '.join(word_list[2:])
    #return accession,region,sci_name,other
    return sci_name

In [14]:
x =prefetch_output['match_name'][1]

In [25]:
std_name(x)

('NC_080369.1',
 '46512-47100',
 'Catolobus pendulus',
 'pendulus isolate Accession 42 chloroplast, complete genome')

In [34]:
prefetch_output['scientific_name'] = prefetch_output['match_name'].apply(lambda x: pd.Series(std_name(x)))

In [36]:
result_df = common_name.merge(prefetch_output, on='scientific_name',how='inner')

In [37]:
result_df.shape

(101, 29)

In [44]:
result_df

Unnamed: 0  food_id                 orig_food_common_name  \
0             0        4                                  Kiwi   
1             4       11                                Cashew   
2            15       16                             Groundnut   
3            18       17                               Burdock   
4            22       19                              Tarragon   
..          ...      ...                                   ...   
96       642760      112                         Biblical Mint   
97       642790      112  Biblical Mint|Hortela da Folha Miuda   
98       647009      145                                Cereza   
99       650333      168                                Savory   
100      653691      196                               Vervain   

                             orig_food_scientific_name  \
0         Actinidia chinensis PLANCHON [Actinidiaceae]   
1            Anacardium occidentale L. [Anacardiaceae]   
2                       Arachis hypogaea L. [Fabaceae]   
3                        Arctium lappa L. [Asteraceae]   
4                Artemisia dracunculus L. [Asteraceae]   
..                                                 ...   
96            Mentha longifolia (L.) HUDS. [Lamiaceae]   
97   Mentha longifolia (L.) HUDS. [Lamiaceae]|Menth...   
98                             Prunus avium [Rosaceae]   
99                     Satureja montana L. [Lamiaceae]   
100               Verbena officinalis L. [Verbenaceae]   

            scientific_name  \
0       Actinidia chinensis   
1    Anacardium occidentale   
2          Arachis hypogaea   
3             Arctium lappa   
4     Artemisia dracunculus   
..                      ...   
96        Mentha longifolia   
97        Mentha longifolia   
98             Prunus avium   
99         Satureja montana   
100     Verbena officinalis   

                                                 other  intersect_bp  \
0                             PLANCHON [Actinidiaceae]          1000   
1                                   L. [Anacardiaceae]          1000   
2                                        L. [Fabaceae]          1000   
3                                      L. [Asteraceae]          1000   
4                                      L. [Asteraceae]          1000   
..                                                 ...           ...   
96                              (L.) HUDS. [Lamiaceae]          1000   
97   (L.) HUDS. [Lamiaceae]|Mentha spicata L. [Lami...          1000   
98                                          [Rosaceae]          1000   
99                                      L. [Lamiaceae]          1000   
100                                   L. [Verbenaceae]          1000   

      jaccard  max_containment  f_query_match  ...  ksize moltype scaled  \
0    0.000562         0.500000       0.500000  ...     31     DNA   1000   
1    0.000563         1.000000       1.000000  ...     31     DNA   1000   
2    0.000563         1.000000       1.000000  ...     31     DNA   1000   
3    0.000563         1.000000       1.000000  ...     31     DNA   1000   
4    0.000563         1.000000       1.000000  ...     31     DNA   1000   
..        ...              ...            ...  ...    ...     ...    ...   
96   0.000563         1.000000       1.000000  ...     31     DNA   1000   
97   0.000563         1.000000       1.000000  ...     31     DNA   1000   
98   0.000562         0.333333       0.333333  ...     31     DNA   1000   
99   0.000563         1.000000       1.000000  ...     31     DNA   1000   
100  0.000562         0.500000       0.500000  ...     31     DNA   1000   

    query_n_hashes  query_abundance query_containment_ani  \
0             1777            False                   NaN   
1             1777            False                   NaN   
2             1777            False                   NaN   
3             1777            False                   NaN   
4             1777            False                   NaN  

In [46]:
result_df.to_csv('prefetch_annot.csv', index=True)

Notes
1. the function std_name has issues seperating accession names.
2. The prefetch results had 5015 matches; Common_name csv has 240 common names. The intersection between these files is around 101. Ie 101 out of 5015 prefetch matches can be annotated by FooDB data. Might have to look into a better database for more coverage.